### DISCUSSION TITLE
Prime Video: "Your browser does not support High-Definition playback due to content restrictions"

### DISCUSSION THREAD

**Discussion by eidleweise** [Floorp-Projects/Floorp#741] [Feature Request]
Floorp 11.8.1 64-bit. I try and watch a video on Amazon Prime and get told "Your browser does not support High-Definition playback due to content restrictions"
![image](https://github.com/Floorp-Projects/Floorp/assets/61235522/f0ae5c68-0c65-4324-9a99-93b48af3facd)

I've found this under vanilla Firefox: https://support.mozilla.org/en-US/questions/1336109 so I assume that it's something that needs to be rolled into this branch?

**Comment by regularentropy** [comment]
> Floorp 11.8.1 64-bit. I try and watch a video on Amazon Prime and get told "Your browser does not support High-Definition playback due to content restrictions" ![image](https://private-user-images.githubusercontent.com/61235522/298195929-f0ae5c68-0c65-4324-9a99-93b48af3facd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU2OTg5NjksIm5iZiI6MTcwNTY5ODY2OSwicGF0aCI6Ii82MTIzNTUyMi8yOTgxOTU5MjktZjBhZTVjNjgtMGM2NS00MzI0LTlhOTktOTNiNDhhZjNmYWNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE5VDIxMTEwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM0NzY1ODIwYmIyNjYzZTUzNTYzZGMxZmRkZjVmOTRhMTdhNGJhNTMxY2U0OThkZjEyM2VhN2RhMTc0YTE2OWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HRAWmEUN_2oDJMHmNx2KUjyJoPAqm47d9wxPRjnPGn0)
> 
> I've found this under vanilla Firefox: https://support.mozilla.org/en-US/questions/1336109 so I assume that it's something that needs to be rolled into this branch?

DRM issue. Related: https://github.com/Floorp-Projects/Floorp/issues/272

**Reply by eidleweise** [reply]
Well that sucks.. so much for Google's "Do no evil"

**Comment by Telmesomething** [comment]
The reason is included in the message namely "content restrictions".  That implies that DRM is not enabled in Floorp. To check, open Floorp Settings and type: DRM in the search box. Toggle "Play DRM-controlled content to On" and click File --> Exit. Restart Floorp and the decryption module should be installed automatically.

**Reply by eidleweise** [reply]
Always worth checking, but sadly "no" 
![image](https://github.com/Floorp-Projects/Floorp/assets/61235522/4074b4ce-591d-45ab-972f-4e605b00ee8f)

From looking around I suspect that this was fixed in an upstream branch of Firefox although I am not sure where Floorp forked. 

This is my current version of Floorp, and I can still reproduce 
![image](https://github.com/Floorp-Projects/Floorp/assets/61235522/6137f6bb-a8d1-4412-bb2f-2471f3a4c879)

**Comment by Telmesomething** [comment]
Open the Plugins link. If DRM is disabled you will see this:
![no drm](https://github.com/Floorp-Projects/Floorp/assets/33827636/a50bf884-d81f-4780-8e9f-eb5a0c8ed153)

If it's enabled, it should look like this:

![drm cdm installed](https://github.com/Floorp-Projects/Floorp/assets/33827636/d09393c3-d2b3-4257-a157-5f0965d85cbb)

Hit Ctrl + Shift + A to jump to the Plugins link.

**Reply by eidleweise** [reply]
Dang you got my hopes up there but yes, it's enabled 
![image](https://github.com/Floorp-Projects/Floorp/assets/61235522/5282ede4-6d69-4614-8d69-7753f18014f9)

**Reply by Telmesomething** [reply]
Pity.

Another thing you can try doing is to disable the Enhanced Tracking Protection which is set to On by default. Access to it by clicking the shield to the left of the URL bar. Also in Settings --> "Privacy and Security" enable defaults if you have any of them set to Strict or something similar. 

Are you logged into a VPN when viewing Amazon stuff? If so, log out. 

I've had similar problems with a site called Zalando which is a German online clothing retailer. If any tracking protections were enabled I couldn't even login. What I eventually did was to create a new profile specifically for that site only with no protections at all including about:config changes which I usually use. After that everything went smoothly. 

If I come across anything else I'll let you know.

**Reply by eidleweise** [reply]
I'm on Standard Security and not on my VPN. The shield for amazon video seems to say that it's happy
![image](https://github.com/Floorp-Projects/Floorp/assets/61235522/1e49bcb6-4277-46c9-bedc-ccb6c1cbae28)

Honestly I think that regularenthropy has the right of it with their comments above https://github.com/Floorp-Projects/Floorp/discussions/741#discussioncomment-8185996. Google are being awkward and until they stop, there's not a lot that Floorp can do about it. Although can we slap google with a wet kipper? Would that help?

**Reply by Telmesomething** [reply]
Slapping Google with a wet kipper sounds very inviting I must say :)

Something else you can try. Go to Tools --> "Show All  History" and find a recent Amazon Prime visit. Right click it and choose "Forget About This Site". That removes everything connected to the Amazon site from Floorp. If there's anything in that data which is causing the problem getting rid of it may fix the problem. Worth a try anyway.

**Reply by Telmesomething** [reply]
Just thought of something else. Click the three dots to the right of the Widevine CDM plugin and make sure it's not set to "Never Activate".

**Reply by eidleweise** [reply]
Never activate gives this error: 
![image](https://github.com/Floorp-Projects/Floorp/assets/61235522/527907d3-ffbb-4a0e-9c8b-427f1dc90e77)

**Reply by eidleweise** [reply]
I also tried forgetting about Amazon and going in again from fresh. Same issue

**Reply by Telmesomething** [reply]
I see what the problem is. It's Floorp, or any Firefox fork for that matter. Firefox may not work either. Error 7235 in your screenshot relates to supported browsers of which [Google Chrome](https://www.amazon.de/-/en/gp/help/customer/display.html?nodeId=G4UPSRVBF5XUKJTA) is the only one.

So slapping Google with a single wet kipper is not enough. It needs a whole truckload of them dropped from a great height!

**Comment by Telmesomething** [comment]
There's a browser called Ungoogled Chrome which replicates the Google original, but without all  the tracking elements. It might be worth a try at least. Just don't set it as the default browser when you install it. I assume you already know that you can have more than one browser on your machine so you can carry on using Floorp for everything else. Here's the [download link](https://github.com/ungoogled-software/ungoogled-chromium).

If it doesn't work you could even install the original Google version using the same principle.

**Comment by SystemError12** [comment]
Same issue here. Amazon Prime Video only work in SD, I have DRM enabled and checked Widevine, still the issue persists. And even worse is Netflix. Netflix won't play any video at all. Not even in SD. 
AppleTV+ on the other hand works fine. 
Something needs to be done to integrate DRM playback properly... .

**Reply by Telmesomething** [reply]
As far as Netflix is concerned Floorp or even Firefox can only be viewed in 720p.  See [Netflix minmum system requirements](https://help.netflix.com/en/node/30081) and go to "System and browser version requirements" on that page.

**Reply by SystemError12** [reply]
As i wrote I cannot play Netflix on Floorp AT ALL. Not even in 720p. Since I am still using the old entry lvl account, I wouldnt even normally get more than that. In Firefox I don't  have any issue, nor on any other browser. Floorp gives me an error code as in the screenshot below: 

![Net](https://github.com/Floorp-Projects/Floorp/assets/156732874/c5467839-c372-4cfb-835c-0ed0e8b1540a)

And then when I click on "Netflix Home" I am instantly logged out of my account and back at log-in screen.

**Reply by Telmesomething** [reply]
From the Netflix support site: [This title is not available to watch instantly](https://github.com/Floorp-Projects/Floorp/discussions/741)

What's your hardware?

**Reply by SystemError12** [reply]
ryzen 9 and rtx 3000 series. Netflix plays on all my other browsers, so I would assume it*s not my hardware.

**Reply by Telmesomething** [reply]
Not being able to watch a Netflix title instantly is a known issue on Netflix. Please see this [link](https://help.netflix.com/en/node/66174)
Click on the link to "All Other Devices" and then click the link to "Contact Us".

**Reply by SystemError12** [reply]
It would help to know if I am the only one with this issue and how other people fare using Netflix with Floorp? As I see it Floorp is not supported by Netflix and that is the possible reason in my opinion. I already looked into it and the error code and at some point it told me my browser is too old and I should update it? Maybe it's because it's based on Firefox ESR and Netflix recognizes a too old version of Firefox?

So yea, I already looked into the error code and Netflix support etc. As wrote above, what really would help me is to know other peoples experiences with Floorp and Netflix? Can anyone reproduce or you guys all have no issues with it?

**Reply by CutterKnife** [reply]
To resolve the Widevine issue, Widevine needs a signature. That can cost a lot of money.
However, Linux is less susceptible to Widevine.

**Reply by breadbeforerice** [reply]
It's pinned in the issue and I hope you'll read it.
https://github.com/Floorp-Projects/Floorp/issues/272

**Reply by SystemError12** [reply]
Oh so it is missing a signature. Thanks for clearing this up. I wonder however what is the difference between Netflix/Amazon (that don't play properly) and AppleTV+ which plays perfectly on Floorp. Just out of curiosity, do they use a different kind of DRM or how come?

**Reply by eidleweise** [reply]
I can confirm that I get the exact same issue when trying to watch Netflix. It feels bad, but I've wound up having to use regular firefox for streaming :(

**Comment by Telmesomething** [comment]
For what it's worth this same subject was posted to the [Amazon Prime forums](https://www.amazonforum.com/s/global-search/Your%20browser%20does%20not%20support%20High-Definition%20playback%20due%20to%20content%20restrictions) three years ago with mixed results

But I would imagine that the Netflix minimum system requirements would apply to Amazon Prime as well.
